src/common/config_parser.cpp

```diff
@@ -377,9 +384,13 @@ void ConfigParser::addOptionsTraining(cli::CLIWrapper& cli) {
 #endif
   // scheduling options
 
+  // In self-adaptive mode users would typically want less updates to happen than in regular training
+  size_t defaultAfterEpochs = (mode_ == cli::mode::selfadaptive) ? 2 : 0;
+  std::string defaultDispFreq = (mode_ == cli::mode::selfadaptive) ? "1" : "1000u";
```

Comment:
For clarity, you can use:
```suggestion
  std::string defaultDispFreq = (mode_ == cli::mode::selfadaptive) ? "1u" : "1000u";
```